packages/x-date-pickers/src/YearPicker/YearPicker.tsx

```diff
@@ -83,29 +66,44 @@ export const YearPicker = React.forwardRef(function YearPicker<TDate>(
     disabled,
     disableFuture,
     disablePast,
-    isDateDisabled,
-    maxDate,
-    minDate,
+    maxDate = defaultProps.maxDate,
+    minDate = defaultProps.minDate,
     onChange,
-    onFocusedDayChange,
-    onYearChange,
     readOnly,
     shouldDisableYear,
   } = props;
 
   const ownerState = props;
   const classes = useUtilityClasses(ownerState);
 
-  const now = useNow<TDate>();
-  const theme = useTheme();
-  const utils = useUtils<TDate>();
-
   const selectedDate = date || now;
   const currentYear = utils.getYear(selectedDate);
   const wrapperVariant = React.useContext(WrapperVariantContext);
   const selectedYearRef = React.useRef<HTMLButtonElement>(null);
   const [focusedYear, setFocusedYear] = React.useState<number | null>(currentYear);
 
+  const isYearDisabled = React.useCallback(
```

Comment:
Copies from #4807

Reply:
I let you add the "Fix #4806" in PR description, and I close #4807?